src/components/carousel/carousel.component.js

```diff
@@ -160,20 +153,18 @@ class BaseCarousel extends React.Component {
       theme: this.props.theme,
       slideProps: {
         className: visibleSlide.props.className,
-        isPadded: this.props.enablePreviousButton || this.props.enableNextButton,
         'data-element': 'visible-slide',
         key: `carbon-slide-${index}`,
         ...visibleSlide.props
       }
     };
-
     return React.cloneElement(visibleSlide, assign({}, visibleSlide.props, additionalProps));
   }
 
   visibleSlides() {
-    const arrayWithKeys = this.props.children.map((element, key) => {
+    const arrayWithKeys = React.Children.map(this.props.children, (element, key) => {
       return React.cloneElement(element, {
-        key: `slide-${key}`,
+        key: `slide-${guid()}`,
```

Comment:
guid's generally shouldn't be used as react keys. To properly fix this, the child needs to have it's own unique identifier, which might be a guid.

Is this something thats broken, if not let's just leave it as the array index, which is the `key` variable here.

Reply:
It was throwing up warnings when running tests and also in the console when the component was loaded. I figured seen as I here doing all this work, I might as well rectify it. I thought it would be something that would be picked up and kicked back to me in QA.

Reply:
We don't get any value from index or guid TBH. React uses them to work out when an element has changed position. If the guid changes on each render, they'll always be replaced. If the index in the array changes, they'll also all be replaced.

Just leave it as guid, it's one less eslint warning and we'll fix it when it's refactored to be a function component.